Added button for collecting user feedback.

barnettwilliam opened the pull request:
Added button for collecting user feedback, when deployed FEEDBAK_SURVEY_URL will be updated to point to the Microsoft forms survey [here](https://forms.office.com/Pages/ResponsePage.aspx?id=FM9wg_MWFky4PHJAcWVDVraRChPUQ-ZGrubhv7wZFeNUM0M4VVQ4SU9MRFVWUUFJSldWU05CUUNCWC4u). 

![image](https://github.com/mdenet/educationplatform/assets/32113205/b5b205ac-2f22-4a50-9914-82cb50d2d9ac)

szschaler commented on platform/public/css/index.css:
```diff
@@ -21,4 +21,20 @@
+    box-shadow: 0 4px 8px 0 gray, 0 6px 20px 0 gray;
+    background-color: lightblue;
+    bottom: 5%;
+    right: 2%;
```
This places this on top of content in the learning activity, which isn't ideal. Could we place it in the bottom-**left** corner instead?

szschaler commented on platform/public/index.html:
```diff
@@ -16,6 +16,12 @@
     <div id="preloader"><img src="images/preloader.gif" width="100px"></div>
 
+    <div id='feedback-button'>
+        <p><a href="FEEDBAK_SURVEY_URL" target="_blank" rel="noopener noreferrer">
```
`FEEDBAK` -> `FEEDBACK` 😄 

More importantly, though: how does this get instantiated effectively? I would like to ensure we don't create more and more places requiring manual adjustment on deployment. Can we automate filling this out on deployment, e.g. via a `.env` file?

barnettwilliam replied:
Corrected in f2298d7605d464f87af539998b569d29fd45ad73 and now settable via en environment variable. The feedback button is now only displayed if the env variable is set.

szschaler reviewed (changes requested):
Generally looks good. 

In the MS Form: you have a nice triaging question at the start. I was expecting this to then lead to different sections so that you can ask specific follow-up questions depending on the type of feedback the student wants to provide.

barnettwilliam commented:
> Generally looks good.
> 
> In the MS Form: you have a nice triaging question at the start. I was expecting this to then lead to different sections so that you can ask specific follow-up questions depending on the type of feedback the student wants to provide.

I have updated the form as suggested to be more selective based on the initial question. For the major issues I added an option to provide the activity for further investigation. I wanted to try to minimise the number of questions asked let me know if there is anything you think specifically would be useful to add, potentially some question on the system they were using e.g. browser, os etc.

szschaler reviewed (approved):
Looks good to me.

In the form, I would suggest keeping the "URL to the platform" link for all types of questions, but make it optional except where issues occurred.

barnettwilliam commented:
Great form updated
